src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.Small.cs

```diff
@@ -67,9 +67,15 @@ protected override object CreateObject(ref ReadStackFrame frame)
 
             bool success = converter.TryRead(ref reader, info.PropertyType, info.Options!, ref state, out TArg? value);
 
-            arg = value == null && jsonParameterInfo.IgnoreNullTokensOnRead
-                ? (TArg?)info.DefaultValue! // Use default value specified on parameter, if any.
-                : value!;
+            if (value == null && jsonParameterInfo.IgnoreNullTokensOnRead)
```

Comment:
I still think that `IsRequired` should override global `IgnoreNullValues` configuration. The latter is marked obsolete so it's impossible for breaking changes to arise here.

Reply:
It's obsolete but I think we should honor it still. IMO not calling Set (or ignoring specific value) is equivalent to not satisfying presence requirement. If we end up adding some new value to JsonIgnoreCondition which acts similarly it would make it more complicated in the future or we'd end up with breaking change

Reply:
Not honoring `required` properties under certain configurations means that the deserializer would be producing instances of types that violate their designated contract, so I think it should always be honored. If silently overriding `IgnoreNullValues` is problematic we can always throw when both `IsRequired` and `IgnoreNullValues` are specified.

@333fred @stephentoub thoughts?

Reply:
I would agree with Eirik. We should either ignore, or inform the user that the model they created has an error.

Reply:
Fair enough, error sounds good to me.